src/platforms/esp32/components/avm_builtins/i2c_driver.c

```diff
@@ -169,12 +187,19 @@ static void i2c_driver_close(Context *ctx)
 {
     struct I2CData *i2c_data = ctx->platform_data;
 
+    MUTEX_LOCK(i2c_data->mutex);
+    ctx->platform_data = NULL;
+
     esp_err_t err = i2c_driver_delete(i2c_data->i2c_num);
     if (UNLIKELY(err != ESP_OK)) {
         ESP_LOGW(TAG, "Failed to delete I2C driver.  err=%i", err);
     }
-    free(ctx->platform_data);
-    ctx->platform_data = NULL;
+
+#ifndef AVM_NO_SMP
+    smp_mutex_destroy(i2c_data->mutex);
```

Comment:
Is this destroying the mutex before unlocking it on purpose? Not sure we can do that, I believe it is undefined behavior with thread APIs such as POSIX.

Reply:
Fixed.